From d392ad204b5241f4929824478c49c52a2b9c87e3 Mon Sep 17 00:00:00 2001
From: Mohamad Noor Alim Hussin <mnalim@efinixinc.com>
Date: Mon, 12 Jan 2026 14:22:29 +0800
Subject: [PATCH] package/libfm: upgrade the version 1.3.2 -> 1.4.0

Add some patches to disable gtk-docs and vala build.

Signed-off-by: Mohamad Noor Alim Hussin <mnalim@efinixinc.com>
---
 package/libfm/0002-remove-building-docs.patch | 50 +++++++++++++++++++
 package/libfm/0003-disable-vala.patch         | 37 ++++++++++++++
 package/libfm/libfm.hash                      |  7 ++-
 package/libfm/libfm.mk                        |  9 ++--
 4 files changed, 95 insertions(+), 8 deletions(-)
 create mode 100644 package/libfm/0002-remove-building-docs.patch
 create mode 100644 package/libfm/0003-disable-vala.patch

diff --git a/package/libfm/0002-remove-building-docs.patch b/package/libfm/0002-remove-building-docs.patch
new file mode 100644
index 0000000000..6c4246b4c2
--- /dev/null
+++ b/package/libfm/0002-remove-building-docs.patch
@@ -0,0 +1,50 @@
+From 40dedee2e35d5281f9bee84e5f49d8ce2b7bc6b0 Mon Sep 17 00:00:00 2001
+From: Mohamad Noor Alim Hussin <mnalim@efinixinc.com>
+Date: Mon, 12 Jan 2026 10:50:36 +0800
+Subject: [PATCH] remove building gtk-doc
+
+Signed-off-by: Mohamad Noor Alim Hussin <mnalim@efinixinc.com>
+---
+ Makefile.am                      | 3 +--
+ docs/reference/libfm/Makefile.am | 4 +---
+ 2 files changed, 2 insertions(+), 5 deletions(-)
+
+diff --git a/Makefile.am b/Makefile.am
+index 427edaeb..0eb9d794 100644
+--- a/Makefile.am
++++ b/Makefile.am
+@@ -8,7 +8,6 @@ DIST_SUBDIRS = src po data
+ SUBDIRS_DOCS = docs
+ 
+ EXTRA_DIST_DOCS = \
+-	gtk-doc.make \
+ 	docs/Makefile.am \
+ 	docs/Makefile.in \
+ 	docs/reference/Makefile.am \
+@@ -48,4 +47,4 @@ pkgconfig_DATA += \
+ 	$(NULL)
+ endif
+ 
+-DISTCHECK_CONFIGURE_FLAGS=--enable-gtk-doc
++DISTCHECK_CONFIGURE_FLAGS=
+diff --git a/docs/reference/libfm/Makefile.am b/docs/reference/libfm/Makefile.am
+index 7747b0b6..284260f2 100644
+--- a/docs/reference/libfm/Makefile.am
++++ b/docs/reference/libfm/Makefile.am
+@@ -127,12 +127,10 @@ GTKDOC_LIBS= \
+ 	$(GTK_LIBS) \
+ 	$(NULL)
+ 
+-# This includes the standard gtk-doc make rules, copied by gtkdocize.
+-include $(top_srcdir)/gtk-doc.make
+ 
+ # Other files to distribute
+ # e.g. EXTRA_DIST += version.xml.in
+-EXTRA_DIST +=
++EXTRA_DIST =
+ 
+ # Files not to distribute
+ # for --rebuild-types in $(SCAN_OPTIONS), e.g. $(DOC_MODULE).types
+-- 
+2.34.1
+
diff --git a/package/libfm/0003-disable-vala.patch b/package/libfm/0003-disable-vala.patch
new file mode 100644
index 0000000000..2cbf4a7686
--- /dev/null
+++ b/package/libfm/0003-disable-vala.patch
@@ -0,0 +1,37 @@
+From e7dbe081083458f713a5edb49103983cc76ffdd8 Mon Sep 17 00:00:00 2001
+From: Mohamad Noor Alim Hussin <mnalim@efinixinc.com>
+Date: Mon, 12 Jan 2026 11:07:53 +0800
+Subject: [PATCH] disable vala
+
+Signed-off-by: Mohamad Noor Alim Hussin <mnalim@efinixinc.com>
+---
+ configure.ac | 6 ++----
+ 1 file changed, 2 insertions(+), 4 deletions(-)
+
+diff --git a/configure.ac b/configure.ac
+index 62988a21..3f06ad4e 100644
+--- a/configure.ac
++++ b/configure.ac
+@@ -36,7 +36,7 @@ AC_ARG_ENABLE(old-actions,
+     [AC_HELP_STRING([--disable-old-actions],
+         [disable old custom actions code if you have no decent Vala compiler])],
+     [enable_actions="${enableval}"],
+-    [enable_actions=yes]
++    [enable_actions=no]
+ )
+ AM_COND_IF(EXTRALIB_ONLY,
+     [enable_actions=no])
+@@ -46,9 +46,7 @@ if test x"$enable_actions" = xyes; then
+         AM_PROG_VALAC
+     else
+         AM_PROG_VALAC([0.13])
+-        if test x"$VALAC" = xvalac; then
+-            AC_ERROR([No Vala compiler found but it is required.])
+-        fi
++	:
+     fi
+     ACTIONS_SUBDIR='actions'
+     AC_DEFINE(HAVE_OLD_ACTIONS, 1, [Define to 1 to enable old custom actions code])
+-- 
+2.34.1
+
diff --git a/package/libfm/libfm.hash b/package/libfm/libfm.hash
index 74781b2a0b..1cab940e65 100644
--- a/package/libfm/libfm.hash
+++ b/package/libfm/libfm.hash
@@ -1,8 +1,7 @@
-# From https://sourceforge.net/projects/pcmanfm/files/PCManFM%20%2B%20Libfm%20%28tarball%20release%29/LibFM/
-md5  c87a0ff41ae77825079b2f785ec0741e  libfm-1.3.2.tar.xz
-sha1  7c5c14dd0615b457687966ce4e55c2d054bf93e1  libfm-1.3.2.tar.xz
+md5  d3147d773aceb53ed6319717a2a0cb9f  1.4.1.tar.gz
+sha1 53f7e5f48f5c61058c4c1cb99240e3f3972ffd94  1.4.1.tar.gz
 
 # Locally computed:
-sha256  a5042630304cf8e5d8cff9d565c6bd546f228b48c960153ed366a34e87cad1e5  libfm-1.3.2.tar.xz
+sha256	1dff59162eb1ca5d1a6b945de4d0bdf2256a5264f54c9225b8599adce45dcb05  1.4.1.tar.gz
 sha256  d8c320ffc0030d1b096ae4732b50d2b811cf95e9a9b7377c1127b2563e0a0388  COPYING
 sha256  c3a3d93e111462f2b0292b56a6a7b87d342fd7156741d8b81b8b08723be4597e  src/extra/fm-xml-file.c
diff --git a/package/libfm/libfm.mk b/package/libfm/libfm.mk
index 419fce59d8..5bb1af6a10 100644
--- a/package/libfm/libfm.mk
+++ b/package/libfm/libfm.mk
@@ -4,16 +4,17 @@
 #
 ################################################################################
 
-LIBFM_VERSION = 1.3.2
-LIBFM_SOURCE = libfm-$(LIBFM_VERSION).tar.xz
-LIBFM_SITE = http://sourceforge.net/projects/pcmanfm/files
+LIBFM_VERSION = 1.4.1
+LIBFM_SOURCE = $(LIBFM_VERSION).tar.gz
+LIBFM_SITE = https://github.com/lxde/libfm/archive/refs/tags
+
 LIBFM_DEPENDENCIES = menu-cache libglib2 cairo libgtk3
 LIBFM_LICENSE = GPL-2.0+, LGPL-2.1+
 LIBFM_LICENSE_FILES = COPYING src/extra/fm-xml-file.c
 LIBFM_INSTALL_STAGING = YES
 # We're patching src/modules/Makefile.am
 LIBFM_AUTORECONF = YES
-LIBFM_CONF_OPTS = --with-gtk=3
+LIBFM_CONF_OPTS = --with-gtk=3 --disable-gtk-doc --disable-old-actions
 
 ifeq ($(BR2_PACKAGE_LIBEXIF),y)
 LIBFM_CONF_OPTS += --enable-exif
-- 
2.43.0

